src/main/scala/sigmastate/types.scala

```diff
@@ -217,8 +225,21 @@ trait STypeCompanion {
   /** List of methods defined for instances of this type. */
   def methods: Seq[SMethod]
 
-  def getMethodById(methodId: Byte): SMethod =
-    methods.find(m => m.objType == this && m.methodId == methodId).get
+  lazy val _methodsMap = methods
+    .groupBy(_.objType.typeId)
+    .map { case (typeId, ms) => (typeId -> ms.map(m => m.methodId -> m).toMap) }
+
+  def hasMethodWithId(methodId: Byte): Boolean =
+    getMethodById(methodId).isDefined
+
+  @inline def getMethodById(methodId: Byte): Option[SMethod] =
```

Comment:
what is the difference between `methodById` and `getMethodById`? Please add docs to these methods describing, why do you need both

Reply:
added description